ddpui/api/org_preferences_api.py

```diff
@@ -0,0 +1,306 @@
+        raise HttpError(404, "Preferences for this organization not found")
+
+    if payload.enable_discord_notifications:
+        if not org_preferences.discord_webhook and not payload.discord_webhook:
+            raise HttpError(400, "Discord webhook is required to enable notifications.")
+        discord_webhook = payload.discord_webhook or org_preferences.discord_webhook
+    else:
+        discord_webhook = None
+
+    org_preferences.enable_discord_notifications = payload.enable_discord_notifications
+    org_preferences.discord_webhook = discord_webhook
+    org_preferences.save()
+
+    response_data = {
+        "enable_discord_notifications": org_preferences.enable_discord_notifications,
+        "discord_webhook": org_preferences.discord_webhook,
+    }
+
+    return {"success": True, "res": response_data}
+
+
+@orgpreference_router.get("/", auth=auth.CustomAuthMiddleware())
+def get_org_preferences(request):
+    """Gets preferences for an organization based on the logged-in user's organization"""
+    orguser: OrgUser = request.orguser
+    org = orguser.org  # Get the organization associated with the current user
+
+    try:
+        # Fetch organization preferences for the user's organization
+        org_preferences = OrgPreferences.objects.select_related("org", "llm_optin_approved_by").get(
+            org=org
+        )
+    except OrgPreferences.DoesNotExist:
+        raise HttpError(404, "Organization preferences not found")
+
+    # Prepare data for the user who approved LLM opt-in, if available
+    approved_by_data = None
+    if org_preferences.llm_optin_approved_by and org_preferences.llm_optin_approved_by.user:
+        user = org_preferences.llm_optin_approved_by.user
+        approved_by_data = {"user_id": user.pk, "email": user.email}
+
+    # Structure the preferences response
+    preferences = {
+        "org": {
+            "id": org_preferences.org.id,
+            "name": org_preferences.org.name,
+            "slug": org_preferences.org.slug,
+            "type": org_preferences.org.type,
+        },
+        "trial_start_date": org_preferences.trial_start_date,
+        "trial_end_date": org_preferences.trial_end_date,
+        "llm_optin": org_preferences.llm_optin,
+        "llm_optin_approved_by": approved_by_data,
+        "llm_optin_date": org_preferences.llm_optin_date,
+        "enable_discord_notifications": org_preferences.enable_discord_notifications,
+        "discord_webhook": org_preferences.discord_webhook,
+    }
+
+    return {"success": True, "res": preferences}
+
+
+@orgpreference_router.get("/org-superset", auth=auth.CustomAuthMiddleware())
+def get_org_superset_details(request):
+    """Gets preferences for an organization by org_id"""
+    orguser: OrgUser = request.orguser
+    org = orguser.org
+
+    try:
+        org_superset = OrgSupersets.objects.select_related(
+            "org",
+        ).get(org=org)
+
+    except OrgSupersets.DoesNotExist:
```

Comment:
why are we doing select_related here ? we already have the org object at line 197

i think we can go away with the try except since we have a global try except in routers file that will catch all generic errors.